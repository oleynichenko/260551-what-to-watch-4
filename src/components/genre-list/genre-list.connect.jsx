import {connect} from "react-redux";
import GenreList from "@components/genre-list/genre-list";
import {getMovieGenres} from "../../reducer/data/selectors";
import {getActiveGenre} from "../../reducer/movies/selectors";
import {ActionCreator as MoviesActionCreator} from "../../reducer/movies/movies";

const mapStateToProps = (state) => ({
  genres: getMovieGenres(state),
  activeGenre: getActiveGenre(state)
});

const mapDispatchToProps = (dispatch) => ({
  onGenreChange(genre) {
    dispatch(MoviesActionCreator.setActiveGenre(genre));
    dispatch(MoviesActionCreator.resetMoviesLimit(genre));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(GenreList);
