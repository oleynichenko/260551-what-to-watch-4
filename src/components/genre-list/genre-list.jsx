const GenreList = (props) => {
  const {genres, activeGenre, onGenreChange} = props;

  return (
    <ul className="catalog__genres-list">
      {genres.map((genre, index) => (
        <li
          key={`${genre}-${index}`}
          className={`catalog__genres-item ${(genre === activeGenre) ? `catalog__genres-item--active` : ``}`}
        >
          <a
            href="#"
            className="catalog__genres-link"
            onClick={(evt) => {
              evt.preventDefault();
              onGenreChange(genre);
            }}
          >{genre}</a>
        </li>
      ))}
    </ul>
  );
};

GenreList.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  activeGenre: PropTypes.string.isRequired,
  onGenreChange: PropTypes.func.isRequired,
};


export default GenreList;
