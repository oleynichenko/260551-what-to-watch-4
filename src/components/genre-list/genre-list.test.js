import {GenreList} from "@components/genre-list/genre-list";

const genres = [`Drama`, `Horror`, `Kids & Family`];

describe(`GenreListComponent`, () => {
  it(`should render GenreList`, () => {
    const tree = renderer.create(
        <GenreList
          genres={genres}
          activeGenre={`Horror`}
          onGenreChange={() => {}}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
