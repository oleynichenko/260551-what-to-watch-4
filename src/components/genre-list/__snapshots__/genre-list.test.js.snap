// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreListComponent should render GenreList 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Drama
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Horror
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Kids & Family
    </a>
  </li>
</ul>
`;
